app-backend/tile/src/main/scala/LayerCache.scala

```diff
@@ -78,17 +82,41 @@ object LayerCache extends Config with LazyLogging {
         }
       }
     )
-  }
 
-  def layerHistogram(layerId: UUID, zoom: Int): OptionT[Future, Array[Histogram[Double]]] = {
+  def layerHistogram(layerId: UUID, zoom: Int): OptionT[Future, Array[Histogram[Double]]] =
     histogramCache.cachingOptionT(s"histogram-$layerId-$zoom") { implicit ec =>
       attributeStoreForLayer(layerId).map { case (store, _) => blocking {
         store.read[Array[Histogram[Double]]](LayerId(layerId.toString, 0), "histogram")
       }}
     }
-  }
 
-  def layerTile(layerId: UUID, zoom: Int, key: SpatialKey): OptionT[Future, MultibandTile] = {
+  def modelLayerHistogram(toolId: UUID, nodeId: Option[UUID]): OptionT[Future, Histogram[Double]] =
+    histogramCache.cachingOptionT(s"model-$toolId-$nodeId") { implicit ec =>
```

Comment:
I think we need to put a cache buster on here (like `tag` with the tile server). That way if a user updates parameters they will not get a stale cache.

Doesn't necessarily have to be dealt with here, I'm ok adding an issue and dealing with this later since this is already a large PR.